bundle/artifacts/upload.go

```diff
@@ -32,9 +37,30 @@ func (m *upload) Apply(ctx context.Context, b *bundle.Bundle) error {
 		return fmt.Errorf("artifact doesn't exist: %s", m.name)
 	}
 
-	if artifact.File == "" {
+	if len(artifact.Files) == 0 {
 		return fmt.Errorf("artifact source is not configured: %s", m.name)
 	}
 
 	return bundle.Apply(ctx, b, getUploadMutator(artifact.Type, m.name))
 }
+
+type cleanUp struct{}
+
+func (m *cleanUp) Name() string {
+	return "artifacts.CleanUp"
+}
+
+func (m *cleanUp) Apply(ctx context.Context, b *bundle.Bundle) error {
+	artifactPath := b.Config.Workspace.ArtifactsPath
+	b.WorkspaceClient().Workspace.Delete(ctx, workspace.Delete{
+		Path:      artifactPath,
+		Recursive: true,
```

Comment:
Not too keen on a recursive delete without validation. If someone configures `artifact_path` to `~` then this would remove everything underneath. Could we do some kind of validation prior to a recursive delete, or better yet, do some kind of tracking which files we have uploaded and only remove those?

Reply:
I'm not a big fan of tracking the uploaded files as it can be a complex and error prone in the end.
Instead I suggest to upload artifacts to subfolder of `artifact_path` and then we can safely delete it each time

Reply:
That works. Agree file tracking would add a lot of complexity.